### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by apricot12**
**In this branch we've followed OOP principles to build a game of Tic-Tac-Toe playable on the CLI with Ruby.**

- **The main user interface file is made executable in bash**

- **Different aspects of the game have their own classes : Board, Game, Player**

### CODE REVIEW

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 
## Well done @team :+1: :100: :1st_place_medal:, you have made an impressive work so far :clap: :clap: :clap: :clap:, but you still have some minor requirements that you need to take care of. check the comments below :point_down: :point_down: :point_down: :point_down: 

- [x] Remove the .gitkepp file from the lib folder.

- [x] Remove all the puts methods from your files other than main.rb.

- [x] You should make sure that user input is valid for example now I can give an empty name to a player and also symbols I can give a space which is invisible, check the image below:
image

always trim your strings with the strip method on Ruby, and make sure they are not empty.

### [OPTIONAL]:

- [ ] it will be super nice if you give the users the choice of quitting the game or playing again.

### Please make all the required changes before you submit another code review.
## Happy Coding :tophat: 
_If you are stuck or have any questions don't hesitate to contact me on slack._

File: bin/main.rb [hunk 1]
```diff
@@ -1,101 +1,86 @@
...
-
-puts "#{player1} is the winner"
-puts "#{player2} is the winner"
-puts "It's a draw!"
+def valid_sign?(input)
+  input.length == 1 && input =~ /[A-Za-z]/ && input != ~ /\s/
+end
+
+def valid_name?(name)
+  name =~ /[A-Za-z]/ && name != ~ /\s/
```
Comment (mcrd25) [inline bin/main.rb:15]: Try to ensure that your check ensures a person cannot enter multiple spaces as they can right now

**Review by mcrd25 — COMMENTED** [review]
Changes Approved ✔️ 

Hi Team!

Good job addressing changes requested. While A person can still enter multiple spaces and have it pass, I don't see it as a hard requirement, which means you get the 👍 

Please do address it first though before moving on.

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

If you liked my review, consider upskilling me on LinkedIn. And please do connect. [My LinkedIn](https://www.linkedin.com/in/mayadouglas/)


*Happy coding 😄,*
**[--Maya Douglas](mayacdouglas.com)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Review by mcrd25 — APPROVED** [review]
I meant to approve before. So It's approved!

**Pull request merged**